src/main/java/net/minecraftforge/client/FancyCloudRenderer.java

```diff
@@ -0,0 +1,435 @@
+        offV = Math.floorMod(offV, texH);
+
+        // Translate the texture.
+        GlStateManager.matrixMode(GL11.GL_TEXTURE);
+        GlStateManager.translate(offU * PX_SIZE, offV * PX_SIZE, 0);
+        GlStateManager.matrixMode(GL11.GL_MODELVIEW);
+
+        GlStateManager.disableCull();
+
+        GlStateManager.enableBlend();
+        GlStateManager.tryBlendFuncSeparate(
+                GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA,
+                GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
+
+        // Color multiplier.
+        Vec3d color = mc.theWorld.getCloudColour(partialTicks);
+        float r = (float) color.xCoord;
+        float g = (float) color.yCoord;
+        float b = (float) color.zCoord;
+
+        if (mc.gameSettings.anaglyph)
+        {
+            float tempR = r * 0.3F + g * 0.59F + b * 0.11F;
+            float tempG = r * 0.3F + g * 0.7F;
+            float tempB = r * 0.3F + b * 0.7F;
+            r = tempR;
+            g = tempG;
+            b = tempB;
+        }
+
+        // Apply a color multiplier through a texture upload if shaders aren't supported.
+        COLOR_TEX.getTextureData()[0] = 255 << 24
+                | ((int) (r * 255)) << 16
+                | ((int) (g * 255)) << 8
+                | (int) (b * 255);
+        COLOR_TEX.updateDynamicTexture();
+
+        GlStateManager.setActiveTexture(OpenGlHelper.lightmapTexUnit);
+        GlStateManager.bindTexture(COLOR_TEX.getGlTextureId());
+        GlStateManager.enableTexture2D();
+
+
+        // Bind the clouds texture last so the shader's sampler2D is correct.
+        GlStateManager.setActiveTexture(OpenGlHelper.defaultTexUnit);
+        mc.renderEngine.bindTexture(TEXTURE);
+
+        ByteBuffer buffer = Tessellator.getInstance().getBuffer().getByteBuffer();
+
+        // Set up pointers for the display list/VBO.
+        if (OpenGlHelper.useVbo())
+        {
+            vbo.bindBuffer();
+
+            int stride = FORMAT.getNextOffset();
+            GlStateManager.glVertexPointer(3, GL11.GL_FLOAT, stride, 0);
+            GlStateManager.glEnableClientState(GL11.GL_VERTEX_ARRAY);
+            GlStateManager.glTexCoordPointer(2, GL11.GL_FLOAT, stride, 12);
+            GlStateManager.glEnableClientState(GL11.GL_TEXTURE_COORD_ARRAY);
+            GlStateManager.glColorPointer(4, GL11.GL_UNSIGNED_BYTE, stride, 20);
+            GlStateManager.glEnableClientState(GL11.GL_COLOR_ARRAY);
+        }
+        else
+        {
+            buffer.limit(FORMAT.getNextOffset());
+            for (int i = 0; i < FORMAT.getElementCount(); i++)
+                FORMAT.getElements().get(i).getUsage().preDraw(FORMAT, i, FORMAT.getNextOffset(), buffer);
+            buffer.position(0);
+        }
+
+        /* Passes:
+        0 - Write depth buffer to prevent insides rendering,
+        1 - Render clouds */
+        for (int pass = 0; pass < (WIREFRAME ? 3 : 2); pass++)
```

Comment:
Why in the world is this a for-loop? Just write the `case` statements one after the other with an if around the 3rd one. This is just... confusing.